File: Android/Java/app/src/main/java/com/microsoft/sampleandroid/SceneformHelper.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.
package com.microsoft.sampleandroid;

import android.Manifest;
import android.app.Activity;
import android.content.Context;
import android.content.pm.PackageManager;
import android.support.v4.content.ContextCompat;
import android.util.Log;

import com.google.ar.core.Config;
import com.google.ar.core.Session;
import com.google.ar.core.exceptions.UnavailableException;
import com.google.ar.sceneform.ArSceneView;

class SceneformHelper {
    private static final String CAMERA_PERMISSION = Manifest.permission.CAMERA;

    // Check to see we have the necessary permissions for this app
    public static boolean hasCameraPermission(Activity activity) {
        return ContextCompat.checkSelfPermission(activity, CAMERA_PERMISSION)
                == PackageManager.PERMISSION_GRANTED;
    }

    public static boolean trySetupSessionForSceneView(Context context, ArSceneView sceneView) {
        try {
            Session session = new Session(context);
            Config config = new Config(session);
            config.setUpdateMode(Config.UpdateMode.LATEST_CAMERA_IMAGE);
            session.configure(config);
            sceneView.setupSession(session);
            return true;
        }
        catch (UnavailableException e) {
            Log.e("ASADemo: ", "Make sure you have a supported ARCore version installed. Exception: " + e.toString());
            return false;
        }
    }
}

File: Xamarin/SampleXamarin.Android/SceneformHelper.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

using Android;
using Android.App;
using Android.Content;
using Android.Content.PM;
using Android.Support.V4.Content;
using Google.AR.Core;
using Google.AR.Sceneform;
using System;

namespace SampleXamarin
{
    internal static class SceneformHelper
    {
        private const string CAMERA_PERMISSION = Manifest.Permission.Camera;

        // Check to see we have the necessary permissions for this app
        public static bool HasCameraPermission(Activity activity)
        {
            return ContextCompat.CheckSelfPermission(activity, CAMERA_PERMISSION)
                    == Permission.Granted;
        }

        public static bool TrySetupSessionForSceneView(Context context, ArSceneView sceneView)
        {
            try
            {
                Session session = new Session(context);
                Config config = new Config(session);
                config.SetUpdateMode(Config.UpdateMode.LatestCameraImage);
                session.Configure(config);
                sceneView.SetupSession(session);
            }
            catch (Exception ex)
            {
                Android.Util.Log.Error("ASADemo: ", ex.ToString());

                return false;
            }

            return true;
        }
    }
}